rmw_zenoh_cpp/src/rmw_test_fixture.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright 2025 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rmw_test_fixture/rmw_test_fixture.h>
+
+#include <cstdlib>
+#include <memory>
+#include <sstream>
+
+#include <nlohmann/json.hpp>
+#include <rcpputils/env.hpp>
+#include <zenoh.hxx>
+#include <zenoh/api/session.hxx>
+
+#include "detail/zenoh_config.hpp"
+
+static std::unique_ptr<zenoh::Session> g_session = nullptr;
+
+static
+std::string
+get_endpoints(zenoh::Session & session)
+{
+  auto zid = session.get_zid();
+  auto keyexpr = zenoh::KeyExpr("@/" + zid.to_string() + "/router");
+  auto replies = session.get(keyexpr, "", zenoh::channels::FifoChannel(2));
+
+  auto reply = replies.recv();
+  const auto & sample = std::get<zenoh::Reply>(reply).get_ok();
+  auto parsed = nlohmann::json::parse(sample.get_payload().as_string());
```

Comment:
Wrap in try-catch block to catch parse errors?

Reply:
Thanks, done in 8a7be3f